docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,518 @@
+                reader.ValueSequence.ToArray() :
+                reader.ValueSpan;
+    // ...
+  }
+}
+```
+
+### Read and write with UTF-8 text
+
+To achieve the best possible performance while using the `Utf8JsonReader` and `Utf8JsonWriter`, read and write JSON payloads already encoded as UTF-8 text rather than as UTF-16 strings. For example, if you're writing string literals, consider caching them as static byte arrays, and write those instead. For a code example, see [Filter data using Utf8JsonReader](system-text-json-how-to.md#filter-data-using-utf8jsonreader).
+
+### Read and write null values
+
+Code that uses the Newtonsoft reader can compare a value token type of String to null, as in the following example:
+
+```csharp
+public static string ReadAsString(this JsonTextReader reader)
+{
+    reader.Read();
+
+    if (reader.TokenType != JsonToken.String)
+    {
+        throw new InvalidDataException();
+    }
+
+    return reader.Value?.ToString();
+}
+```
+
+This code doesn't work correctly in `System.Text.Json`, which considers the `null` literal to be its own token type. When using `Utf8JsonReader`, check for the `JsonTokenType.Null` token type, as shown in the following example:
+
+```csharp
+public static string ReadAsString(this ref Utf8JsonReader reader)
+{
+    reader.Read();
+
+    if (reader.TokenType == JsonTokenType.Null)
+    {
+        return null;
+    }
+
+    if (reader.TokenType != JsonTokenType.String)
+    {
+        throw new InvalidDataException();
+    }
+
+    return reader.GetString();
+}
+```
+
+To write null values by using `Utf8JsonWriter`, call:
+
+* <xref:System.Text.Json.Utf8JsonWriter.WriteNull%2A> to write a key-value pair with null as the value.
+* <xref:System.Text.Json.Utf8JsonWriter.WriteNullValue%2A> to write null as an element of a JSON array.
+
+For a string property, if the string is null, <xref:System.Text.Json.Utf8JsonWriter.WriteString%2A> and <xref:System.Text.Json.Utf8JsonWriter.WriteStringValue%2A> are equivalent to `WriteNull` and `WriteNullValue`.
+
+### Multi-targeting
+
+If you need to continue to use Newtonsoft for certain target frameworks, you can multi-target and have two implementations. However, this is not trivial and would require some `#ifdefs` and source duplication. One way to share as much code as possible is to create a `ref struct` wrapper around `Utf8JsonReader` and Newtonsoft `JsonTextReader`, and a wrapper around `Utf8JsonWriter` and `JsonTextWriter`. This wrapper would unify the public surface area while isolating the behavioral differences. This lets you isolate the changes mainly to the construction of the type, along with passing the new type around by reference. This is the pattern that the [.NET Core installer](https://github.com/dotnet/runtime/tree/master/src/installer) follows:
+
+* [UnifiedJsonReader.JsonTextReader.cs](https://github.com/dotnet/runtime/blob/master/src/installer/managed/Microsoft.Extensions.DependencyModel/UnifiedJsonReader.JsonTextReader.cs)
```

Comment:
These should point to an exact hash. Pointing to the master branch is problematic if the types move around or is deleted